enterprise/frontend/src/embedding-sdk/cli/steps/pick-database-tables.ts

```diff
@@ -0,0 +1,86 @@
+import { checkbox } from "@inquirer/prompts";
+import ora from "ora";
+
+import type { CliStepMethod } from "embedding-sdk/cli/types/cli";
+import type { Table } from "metabase-types/api";
+
+import { propagateErrorResponse } from "../utils/propagate-error-response";
+import { retry } from "../utils/retry";
+
+export const pickDatabaseTables: CliStepMethod = async state => {
+  const { instanceUrl, databaseId, cookie = "" } = state;
+
+  const spinner = ora("Fetching database schemas and tables…").start();
+  let schemas: string[] = [];
+
+  // Fetch the database schemas
+  try {
+    schemas = await retry(
```

Comment:
We retry until we've found the schemas - as the schema takes a bit of time to sync.